packages/drivers/odsp-driver/src/odspCache.ts

```diff
@@ -146,3 +150,11 @@ export interface IPrefetchSnapshotContents extends ISnapshotContents {
 	fluidEpoch: string;
 	prefetchStartTime: number;
 }
+
+/**
+ * @alpha
+ */
+export interface IPrefetchPartialSnapshotWithContents extends IPartialSnapshotWithContents {
```

Comment:
There are two ways to avoid this kind of changes:
1. Keep ISnapshotContents & IPartialSnapshotWithContents closer to each other.
2. Instead of using ISnapshotContents in code, define new type (ISnapshotContents | IPartialSnapshotWithContents) and use it instead with all constructs. You would be able to use "extends" flow (i.e. IPrefetchSnapshotContents would not be able to extend from ISnapshotContents), but I think you can use equivalent:

```ts
type IPrefetchSnapshotContents = (ISnapshotContents | IPartialSnapshotWithContents ) & {
	fluidEpoch: string;
	prefetchStartTime: number;
}
```

Not sure if it's much better, but it avoids proliferation of extra types. I'd prefer types to be extension of each other to avoid it.